test/widgets/profile_test.dart

```diff
@@ -0,0 +1,269 @@
+        realmDefaultExternalAccounts: {
+          'external1': RealmDefaultExternalAccount(
+            name: 'external1',
+            text: '',
+            hint: '',
+            urlPattern: 'http://example/%(username)s')},
+      );
+
+      await tester.tap(find.text('externalValue'));
+      final expectedMode = defaultTargetPlatform == TargetPlatform.android ?
+        LaunchMode.externalApplication : LaunchMode.platformDefault;
+      check(testBinding.takeLaunchUrlCalls())
+        .single.equals((url: Uri.parse('http://example/externalValue'), mode: expectedMode));
+    });
+
+    testWidgets('page builds; user links to profile', (WidgetTester tester) async {
+      final users = [
+        eg.user(userId: 1, profileData: {
+          0: ProfileFieldUserData(value: '[2]'),
+        }),
+        eg.user(userId: 2, fullName: 'test user'),
+      ];
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+      await setupPage(tester,
+        users: users,
+        pageUserId: 1,
+        customProfileFields: [mkCustomProfileField(0, CustomProfileFieldType.user)],
+        navigatorObserver: testNavObserver,
+      );
+
+      final textFinder = find.text('test user');
+      check(textFinder.evaluate()).length.equals(1);
+      final fieldContainer = find.ancestor(of: textFinder, matching: find.byType(Column)).first;
+      final targetWidget = find.descendant(of: fieldContainer, matching:find.byType(Avatar));
+      await tester.tap(targetWidget, warnIfMissed: false);
+      check(pushedRoutes).last.isA<WidgetRoute>().page.isA<ProfilePage>().userId.equals(2);
+    });
+
+    testWidgets('page builds; dm links to correct narrow', (WidgetTester tester) async {
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      final expectedNarrow = DmNarrow(
+        allRecipientIds: [eg.selfUser.userId, 1]..sort(),
+        selfUserId: eg.selfUser.userId);
+
+      await setupPage(tester,
+        users: [eg.user(userId: 1)],
+        pageUserId: 1,
+        navigatorObserver: testNavObserver,
+      );
+
+      final targetWidget = find.byIcon(Icons.email);
+      await tester.tap(targetWidget, warnIfMissed: false);
+      check(pushedRoutes).last.isA<WidgetRoute>().page
+        .isA<MessageListPage>()
+        .narrow.equals(expectedNarrow);
+    });
+
+    testWidgets('page builds; user links render multiple avatars', (WidgetTester tester) async {
+      final users = [
+        eg.user(userId: 1, profileData: {
+          0: ProfileFieldUserData(value: '[2,3]'),
+        }),
+        eg.user(userId: 2, fullName: 'test user1'),
+        eg.user(userId: 3, fullName: 'test user2'),
```

Comment:
Let's also have a test where there's an unknown user — mostly just to confirm we don't crash in that case, but also that we do show "(unknown user)" rather than silently drop them from the list.

See test/widgets/recent_dm_conversations_test.dart for a couple of examples.